src/app/content/utils.ts

```diff
@@ -42,3 +48,24 @@ export const preloadedPageIdIs = (window: Window, id: string) => window.__PRELOA
   && window.__PRELOADED_STATE__.content
   && window.__PRELOADED_STATE__.content.page
   && window.__PRELOADED_STATE__.content.page.id === id;
+
+export const mapBookSections = (book: Book) => {
+  return new Map(flattenArchiveTree(book.tree).filter((section) =>
+    (section.parent && archiveTreeSectionIsBook(section.parent))
+    || archiveTreeSectionIsChapter(section)).map((s) => [s.id, s]));
+};
+
+export const getCurrentChapter = (sections: BookSections, page: Page) => {
```

Comment:
since this is acting on a given page, not necessarily the _current_ page, i'd call it `getPageChapter`, but this isn't actually getting the page chapter, its getting the highlighting filter option for a page, so it should probably be named something else and moved into the highlighting module.

it would be nice to have a generic `getPageChapter` function though, i think we could use that elsewhere

Reply:
I've renamed it to `getHighlightLocationForPage` and moved it to highlight utils